src/pages/GamerPassport/ActionModals.tsx

```diff
@@ -0,0 +1,343 @@
+                  to our{" "}
+                  <a className="!text-[#7a98df] hover:underline" href="https://itheum.io/discord" target="blank">
+                    discord
+                  </a>{" "}
+                  and raise the issue and once you prove ownership of your PSN account, we can transfer ownership to you! Or, if you mistyped your username,{" "}
+                  <span
+                    className="text-[#7a98df] hover:underline cursor-pointer"
+                    onClick={() => {
+                      setEligibilityCheckStepDone(false);
+                      setEligibilityUserNameTaken(false);
+                      setEligibilityUserNameNotSuitable(false);
+                    }}>
+                    try again here
+                  </span>
+                </p>
+              </>
+            )}
+
+            {eligibilityCheckStepDone && eligibilityUserNameNotSuitable && (
+              <>
+                <FontAwesomeIcon fade={true} color="#4691e2" icon={faHeartCrack} size="3x" className="m-2" />
+                <p className="text-lg font-bold">That username is not suitable!</p>
+                <p className="mt-2">
+                  PSN username <span className="text-[#4691e2]">{psnUserName}</span> cannot be found or does not produce any gaming data that we can use. Are
+                  you sure you got the username right? if so, head over to our{" "}
+                  <a className="!text-[#7a98df] hover:underline" href="https://itheum.io/discord" target="blank">
+                    discord
+                  </a>{" "}
+                  and raise the issue and we can investigate! Or, if you mistyped your username,{" "}
+                  <span
+                    className="text-[#7a98df] hover:underline cursor-pointer"
+                    onClick={() => {
+                      setEligibilityCheckStepDone(false);
+                      setEligibilityUserNameTaken(false);
+                      setEligibilityUserNameNotSuitable(false);
+                    }}>
+                    try again here
+                  </span>
+                </p>
+              </>
+            )}
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export const ActionModalStep3 = (props: any) => {
+  const { publicKey, signMessage } = useWallet();
+  const { showActionModel, handleHideActionModel } = props;
+  const { step1PSNUserName, step1PSNEligibilityCheckResults, step2SolanaAddress, handleStep3Passed } = props;
+
+  const [userSaveToLogAttempted, setUserSaveToLogAttempted] = useState<boolean>(false);
+  const [userSavedToLog, setUserSavedToLog] = useState<boolean>(false);
+  const [userSavedToLogError, setUserSavedToLogError] = useState<boolean>(false);
+  const [saveToLogLoading, setSaveToLogLoading] = useState<boolean>(false);
+
+  async function saveUserToLogViaAPI() {
+    setSaveToLogLoading(true);
+
+    // get signature
+    const preAccessNonce = await itheumSolPreaccess();
```

Comment:
This could be improved by using the caching of the preaccess signature. Could make some users skip signing messages. See example in the NFTunes view data.

Not mandatory, can also be added later

Reply:
will implement in next release